observatory-platform/observatory/platform/telescopes/organisation_telescope.py

```diff
@@ -0,0 +1,167 @@
+# Copyright 2020 Curtin University
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Author: Aniek Roelofs
+
+from typing import Dict, List
+
+import pendulum
+from google.cloud import bigquery
+from google.cloud.bigquery import SourceFormat
+
+from observatory.api.client.model.organisation import Organisation
+from observatory.platform.telescopes.telescope import Release, Telescope
+from observatory.platform.utils.airflow_utils import AirflowVars
+from observatory.platform.utils.template_utils import (blob_name, bq_load_partition, table_ids_from_path,
+                                                       upload_files_from_list)
+
+
+class OrganisationRelease(Release):
+    def __init__(self, dag_id: str, release_date: pendulum.Pendulum, organisation: Organisation,
+                 download_files_regex: str = None, extract_files_regex: str = None, transform_files_regex: str = None):
+        """ Construct an OrganisationRelease instance.
+
+        :param dag_id: the id of the DAG.
+        :param release_date: the release date (used to construct release_id).
+        :param organisation: the Organisation created with the Observatory API.
+        :param download_files_regex: regex pattern that is used to find files in download folder
+        :param extract_files_regex: regex pattern that is used to find files in extract folder
+        :param transform_files_regex: regex pattern that is used to find files in transform folder
+        """
+        self.release_date = release_date
+        release_id = f'{dag_id}_{self.release_date.strftime("%Y_%m_%d")}'
+        super().__init__(dag_id, release_id, download_files_regex, extract_files_regex, transform_files_regex)
+        self.organisation = organisation
+
+    @property
+    def download_bucket(self):
+        """ The download bucket name.
+
+        :return: the download bucket name.
+        """
+        return self.organisation.gcp_download_bucket
+
+    @property
+    def transform_bucket(self):
+        """ The transform bucket name.
+
+        :return: the transform bucket name.
+        """
+        return self.organisation.gcp_transform_bucket
+
+
+class OrganisationTelescope(Telescope):
```

Comment:
Are these types of telescopes structurally different enough from the other two templates to justify its own template? from the earlier tutorial description it seems like it's just some of the variables are constructed differently (from api data).  Another approach could be to have library functions that help handle those kinds of things instead.

Reply:
I would argue that is quite different, for the telescope class itself it always needs the Organisation parameter, has different bucket properties and a distinct bigquery load method. 
Creating a template for this saves some duplicated code in each of the mellon telescopes, I don't think those duplicated bits can be handled by library functions.

Another reason that I've added this template is because it makes it a bit easier to generate the correct template files for a telescope that needs the organisation from the observatory API.
Because it is a distinct telescope type, it is possible to create a DAG file template that uses the observatory API and a unit test template that is set-up to use the observatory API.